green_cli/liquid.py

```diff
@@ -43,6 +47,12 @@ def getassetinfo(session, details):
     details['assets'] = True
     return session.refresh_assets(details)
 
+@functools.lru_cache(maxsize=None)
+def _get_assets_by_name(session):
+    """Get asset registry indexed by name."""
+    asset_info = _get_assets(session)
+    return {details['name'] : details for asset, details in asset_info['assets'].items()}
```

Comment:
The asset registry does not enforce "name" uniqueness.
For instance we have (more than) 2 assets with name "Test", https://assets.blockstream.info/2d53168600a710a103bfef8615fe613385da7ad5cc7f14f5039ba5b59b78727d, https://assets.blockstream.info/61fbbb65680bd2bf64de7543e4838d5c785b3720e3c1e996ad2f5e29cc04bec9.
ATM it just enforces uniqueness of "ticker-domain", but that might be relaxed in the future, thus I would not rely on that.

So the above is overwriting assets with duplicate names.

I think we should just rely on the `asset_id` and perhaps give a special treatment to "btc".
Maybe in the future we want to do something like assetdir in Elements Core, but I reckon we need some kind of confirmation by the user before allowing to use the name or ticker for issued assets.

Reply:
I will update to allow mapping of all unique names and ignore non-unique ones.

Allowing any duplicates across the set of ID's, names and tickers is clearly a recipe for disaster since we don't control the UX the user will use to select assets. This should be fixed in the registry without delay.

Reply:
Updated